internal/stores/s3.go

```diff
@@ -180,7 +180,7 @@ func (s *S3Store) Retrieve(ctx context.Context, objects ...string) error {
 		objectPath := strings.TrimSuffix(o, filepath.Ext(o))
 		// We will either read the file that already exists or download it because it
 		// is missing
-		f, err := openOrCreateFile(objectPath)
+		f, err := openOrCreateFile(s.fsys, objectPath)
```

Comment:
++, this is good, allows the use of many types of afero backends like `InMem` or `ReadOnlyOsFs`